src/metadata/Metadata.ts

```diff
@@ -297,10 +299,113 @@ export class Metadata extends Instantiable {
         return result
     }
 
+    /**
+     * Retrieves a service by its agreementId.
+     * @param  {string} agreementId agreementId of the service.
+     * @return {Promise<ServiceCommon>} Service object.
+     */
+    public async retrieveService(
+        agreementId: string,
+        metadataServiceEndpoint?: string
+    ): Promise<ServiceCommon> {
+        const fullUrl =
+            metadataServiceEndpoint || `${this.url}${servicePath}/${agreementId}`
+        const result = await this.nevermined.utils.fetch
+            .get(fullUrl)
+            .then((response: any) => {
+                if (response.ok) {
+                    return response.json()
+                }
+                this.logger.log(
+                    'retrieveService failed:',
+                    response.status,
+                    response.statusText,
+                    agreementId
+                )
+                return null as ServiceCommon
+            })
+            .then((response: ServiceCommon) => {
+                return response as ServiceCommon
+            })
+            .catch(error => {
+                this.logger.error('Error retrieving service: ', error)
+                return null as ServiceCommon
+            })
+
+        return result
+    }
+
+    /**
+     *
+     * @param agreement stores the Service object with its agreementId as
+     * @returns the newly stored service object
+     */
+    public async storeService(
+        agreementId: string,
+        agreement: ServiceCommon
+    ): Promise<ServiceCommon> {
+        const fullUrl = `${this.url}${servicePath}`
+        agreement['agreementId'] = agreementId
+        const result: ServiceCommon = await this.nevermined.utils.fetch
+            .post(fullUrl, JSON.stringify(agreement))
+            .then((response: any) => {
+                if (response.ok) {
+                    return response.json()
+                }
+                this.logger.error(
+                    'storeService failed:',
+                    response.status,
+                    response.statusText,
+                    agreement
+                )
+                return null as DDO
+            })
+            .then((response: ServiceCommon) => {
+                return response as ServiceCommon
+            })
+            .catch(error => {
+                this.logger.error('Error storing service: ', error)
+                return null as ServiceCommon
+            })
+        console.log('SaveService: ', result)
+        return result
+    }
+
     public getServiceEndpoint(did: DID) {
         return `${this.url}${apiPath}/did:nv:${did.getId()}`
     }
 
+    public async retrieveServiceAgreement(
```

Comment:
what is the difference between this method and `retrieveService`?

Reply:
Nothing, came from Yash's branch. Deleting this.